### ISSUE TITLE
How to watch the capture worksheet/cell change events?

### ISSUE THREAD

**Issue by lapd-viosoft** [ONLYOFFICE/sdkjs-plugins#130]
Hi,
Thank for the great project.
As we know, Microsoft Excel and OfficeJs addins allow us to listen event when a worksheet and cell change event.
Is there anyway to listen these change events with OnlyOffice via plugin/macro?
Thank you!

**Comment by ShockwaveNN** [comment]
@flaminestone Please, take a look

**Comment by flaminestone** [comment]
Hello @lapd-viosoft.This feature is in development and has not been documented yet.
What you mean with worksheet change event? In spreadsheet editor, every cell change (or double click in any cell) will trigger this event, and I do not know any case for using this.

**Comment by lapd-viosoft** [comment]
Hello @flaminestone 
In Excel, with VBA macro, we can watch the change of each cell and do further processing. For example:
```vba
Private Sub Worksheet_Change(ByVal Target As Excel.Range) 
    If Target.Column = 1 Then 
        ThisRow = Target.Row 
        If Target.Value > 100 Then 
            Range("B" & ThisRow).Interior.ColorIndex = 3 
        Else 
            Range("B" & ThisRow).Interior.ColorIndex = xlColorIndexNone 
        End If 
    End If 
End Sub
```

Or with OfficeJs addins, we can do like this
```js

// Register change event
Excel.run(function (context) {
    var worksheet = context.workbook.worksheets.getItem("Sample");
    worksheet.onChanged.add(handleChange);

    return context.sync()
        .then(function () {
            console.log("Event handler successfully registered for onChanged event in the worksheet.");
        });
}).catch(errorHandlerFunction);


function handleChange(event)
{
    return Excel.run(function(context){
        return context.sync()
            .then(function() {
                console.log("Change type of event: " + event.changeType);
                console.log("Address of event: " + event.address);
                console.log("Source of event: " + event.source);
   
            });
    }).catch(errorHandlerFunction);
}

```

> In spreadsheet editor, every cell change (or double click in any cell) will trigger this event, and I do not know any case for using this.
Can you tell me how to capture this cell change event in plugin/macro or any background thread of editor?

I have scenario like: 
- A plugin as system plugin, run in background and listen the change of every cell. 
- When a cell is changed (ex: user input some value, or paste new value, or select new value from drop down list) , the plugin captures new content, check if the new content again some rules....and give warning to user if the new value is invalid.

Thank you.

**Comment by flaminestone** [comment]
> Can you tell me how to capture this cell change event in plugin/macro or any background thread of editor?


It is not available now, this functional is in development.

**Comment by lapd-viosoft** [comment]
Hi @flaminestone 
Thank you very much for your information!

**Comment by gamead** [comment]
Is it available for now?

**Comment by l8556** [comment]
Hi @gamead ! 
These methods are still in development.
Improvement request in our internal repository #58189

**Comment by trofim24** [comment]
@Starman-in-the-sky yes, we've added `onWorksheetChange` event and you can attach it with [attachEvent](https://api.onlyoffice.com/docs/office-api/usage-api/spreadsheet-api/Api/Methods/attachEvent/) and use it in [Automation API](https://api.onlyoffice.com/docs/docs-api/usage-api/automation-api/) or [macros](https://api.onlyoffice.com/docs/plugin-and-macros/macros/getting-started/) . Description for all events is in progress.
Using this event in plugins we're adding in next version
